src/main/kotlin/io/openfuture/state/blockchain/EthereumBlockchain.kt

```diff
@@ -28,12 +28,19 @@ class EthereumBlockchain(private val web3j: Web3j) : Blockchain() {
         return UnifiedBlock(transactions, date, block.number.toLong(), block.hash)
     }
 
-    private fun obtainTransactions(ethBlock: EthBlock.Block): List<UnifiedTransaction> = ethBlock.transactions
+    private suspend fun obtainTransactions(ethBlock: EthBlock.Block): List<UnifiedTransaction> = ethBlock.transactions
             .map { it.get() as EthBlock.TransactionObject }
-            .filter { null != it.from && null != it.to }
             .map { tx ->
+                val address = tx.to ?: findContractAddress(tx.hash)
                 val amount = Convert.fromWei(tx.value.toBigDecimal(), Convert.Unit.ETHER)
-                UnifiedTransaction(tx.hash, tx.from, tx.to, amount)
+                UnifiedTransaction(tx.hash, address, amount)
             }
 
+    private suspend fun findContractAddress(transactionHash: String) = web3j.ethGetTransactionReceipt(transactionHash)
+            .sendAsync()
```

Comment:
I believe you can put `sendAsync` and `await` on the same line.

As well as `transactionReceipt` and `get()`